mod-kitpvp/src/main/java/gg/solarmc/loader/kitpvp/KitPvp.java

```diff
@@ -123,6 +126,87 @@ public StatisticResult addAssists(Transaction transaction, int amount) {
     }
 
     /**
+
+     * Adds experience to the user account. Infallible.
+     * @param transaction represents the transaction
+     * @param amount represents the amount of xp to add
+     * @return a result with the new value of xp
+     */
+    public StatisticResult addExperience(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingValue = statisticsRecord.getExperience();
+        int newValue = existingValue + amount;
+
+        statisticsRecord.setExperience(newValue);
+        statisticsRecord.store(KITPVP_STATISTICS.EXPERIENCE);
+
+        this.updateExperience(newValue);
+
+        return new StatisticResult(newValue);
+    }
+
+    /**
+     * Adds current killstreaks to the user account. Infallible.
+     *
+     * This will increment both current and highest killstreaks.
+     *
+     * @param transaction represents the current killstreak
+     * @param amount represents the amount of killstreak to add
+     * @return a result with the new value of killstreak
+     */
+    public StatisticResult addKillstreaks(Transaction transaction, int amount) {
+        if (amount <= 0) {
+            throw new IllegalArgumentException("amount must be positive");
+        }
+        KitpvpStatisticsRecord statisticsRecord = getStatistics(transaction);
+
+        int existingKillstreaks = statisticsRecord.getCurrentKillstreak();
+        int newKillstreaks = existingKillstreaks + amount;
+
+        int existingHighKillstreaks = statisticsRecord.getHighestKillstreak();
+
+        if (newKillstreaks > existingHighKillstreaks) {
+            statisticsRecord.setHighestKillstreak(newKillstreaks);
+            this.updateHighestKillstreak(newKillstreaks);
+            statisticsRecord.store(KITPVP_STATISTICS.HIGHEST_KILLSTREAK);
+        }
+
+        statisticsRecord.setCurrentKillstreak(newKillstreaks);
+        statisticsRecord.store(KITPVP_STATISTICS.CURRENT_KILLSTREAK);
+
+        this.updateCurrentKillstreak(newKillstreaks);
+
+        return new StatisticResult(newKillstreaks);
+    }
+
+    /**
+     * Sets the current killstreak amouont. Infallible
+     *
+     * Used to reset the current killstreak
+     *
+     * @param transaction represents the current transaction
+     * @param amount represents the amount of killstreak to set to
+     */
+    public void setCurrentKillstreaks(Transaction transaction, int amount) {
```

Comment:
This method updates the current killstreak to the specified amount, but it no longer updates the highest killstreak. That is unadvisable since it could lead to the highest killstreak becoming out-of-sync with the current killstreak.

If all you want to do is reset the current killstreak, then you should make a method `resetCurrentKillstreak` which sets the killstreak to 0.